File: 백준/Silver/12789. 도키도키 간식드리미/도키도키 간식드리미.py
import sys
input = sys.stdin.readline

n = int(input())
line = list(map(int,input().split()))
stack=[]
chk=1

for i in line:
    stack.append(i)
    while stack and stack[-1] == chk:
        stack.pop()
        chk += 1
    if len(stack) > 1 and stack[-1] > stack[-2]:
        print("Sad")
        sys.exit()
if stack:
	print("Sad")
else:
	print("Nice")

File: 백준/Silver/12789. 도키도키 간식드리미/도키도키 간식드리미.cc
#include <stdio.h>

#define MAX_N 1001

int main() {
    int n;
    int line[MAX_N];
    int stack[MAX_N];
    int top = -1;
    int chk = 1;

    scanf("%d", &n);
    for (int i = 0; i < n; i++) {
        scanf("%d", &line[i]);
    }

    for (int i = 0; i < n; i++) {
        stack[++top] = line[i];
        
        while (top >= 0 && stack[top] == chk) {
            top--;
            chk++;
        }
        
        if (top > 0 && stack[top] > stack[top - 1]) {
            printf("Sad\n");
            return 0;
        }
    }

    if (top >= 0) {
        printf("Sad\n");
    } else {
        printf("Nice\n");
    }

    return 0;
}